systemtest/gencorpora/apmserver.go

```diff
@@ -0,0 +1,164 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gencorpora
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/elastic/apm-server/systemtest/apmservertest"
+)
+
+// APMServer represents a wrapped exec CMD for APMServer process
+type APMServer struct {
+	apmHost string
+	cmd     *apmservertest.ServerCmd
```

Comment:
Is there a reason why you're not using apmservertest.Server?

Reply:
I read a little further and probably answered my own question: apmservertest.Server requires a `testing.TB`. Maybe we can get rid of that requirement?

Reply:
That was my first approach but I decided to abandon it because the `apmservertest.Server` is tightly coupled with `testing.TB` utilizing `Cleanup` function to prevent unnecessary handling to leak into the test cases. I was thinking to extend the `APMServer` command to meet the requirements of test cases in a generic way and then replace `apmservertest#Server` to use it with `testing.TB`. But, this proved to be a lot more work than I anticipated and I thought we can address it at a later point. Let me know if you think it should be addressed now.

Reply:
Yes, that's fair enough.

I spiked on making the `testing.TB` part optional - I might create a followup on your PR if you don't mind.